pkg/virtctl/imageupload/imageupload.go

```diff
@@ -84,6 +90,9 @@ type HTTPClientCreator func(bool) *http.Client
 
 var httpClientCreatorFunc HTTPClientCreator
 
+// UploadProcessingCompleteFunc the function called while determining if post transfer processing is complete.
+var UploadProcessingCompleteFunc = waitUploadProcessingComplete
```

Comment:
There really isn't an API at all, only reason I did it this way to easily stub out the function in the tests because the tests don't update the status of the PVC and the real function will essentially wait forever for the status to update.

Reply:
There is an interface here... even if it's used only for stubbing, it exists and personally, I'd prefer to be explicit here about what the type of `UploadProcessingCompleteFunc` is where we declare... I think that it makes the code more readable in this case.

Reply:
I completely disagree, defining a type, make a separate variable and then having to check if it is assigned is more code, less readable and gains us nothing. But if that what it takes to get this approved, fine.

Reply:
You don't have to assign it separately ... you can assign it right where it was before but just with the type defined explicitly.
Anyway, I had no intentions to be picky in the review or something, this is an open discussion and you can disagree with me and be sure that I won't block your PR from being merged because of something like this... this suggestion is 100% a matter of taste.

Reply:
I see what you mean, I modified the code. I still think its a little strange, but that works for me.

Reply:
Thanks for updating